### ISSUE TITLE
Loss should change depending on the number of epochs chosen even if I set the seed?

### ISSUE THREAD

**Issue by PabloRR100** [tensorflow/tensorflow#15235]
Hi colleagues,

I am using Keras to train different NNs. I would like to know why if I increment the epochs, the result until the same number of epochs is not the same for the evolution of loss. I am using shuffle=False, and np.random.seed(2017), tf.set_random_seed(2017), and I have check that if I repeat with the same number of epochs, the result is the same, so the random initialization is working. After the epochs training, I am deleting the ANN so the training begins at 0 again.

Here I attach the picture of the resulting training with 10 epochs:

<img width="676" alt="captura de pantalla 2017-12-09 a las 15 02 25" src="https://user-images.githubusercontent.com/23745991/33796312-05efe5da-dcf2-11e7-9780-09e0be902557.png">

And here I attach the picture of the resulting training with 8 epochs:

<img width="679" alt="captura de pantalla 2017-12-09 a las 15 02 34" src="https://user-images.githubusercontent.com/23745991/33796313-106a6b34-dcf2-11e7-820c-e5d279665785.png">

Also, I would like to know why the training time is not exactly (8/10) the 10 epochs attempt and how is it possible that some of them have less accuracy with 2 more epochs!

Here is the link to open code Jupyter Notebook. [GitHub Jupyter Notebook - ANN Comparison](https://github.com/PabloRR100/Pruning-Algorithm-Method/blob/master/NN-Comparison.ipynb)

Thanks a lot!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by PabloRR100** [comment]
I am sorry, here I attached the required information

### System information
OS: MacOs High Sierra 10.13.1

Anaconda Environment
- Python 3.5.4
- Tensorflow installed via pip command from Tensorflow website
- Tensorflow version: v1.4.0-rc1-11-g130a514 1.4.0
- No CUDA. GPU = Intel HD Graphics 530 1536 MB

I hope this is all the info you need. If not, please let me know,

Thanks!

**Comment by angersson** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
